### PR TITLE
Passes the parsed new value to the updateSchema call in updatePropert…

### LINKED ISSUE #1158: When calling updateSchema we are passing the current attribute value and this can be null
We are also parsing the attribute value through the styleParser. The ```updateSchema``` will probably expect the data parsed through the previous schema.

### PR DESCRIPTION
**Pull request by dmarcos**
Changes proposed:
-
-
-

…ies (fixes #1158)

### CODE REVIEW

**Comment by dmarcos** [comment]
Fixes the pano-explorer example

File: tests/core/component.test.js [hunk 1]
```diff
@@ -225,4 +225,30 @@ suite('Component', function () {
...
   });
+
+  suite('updateProperties', function () {
+    setup(function () {
+      components.dummy = undefined;
+    });
+
+    test('updates the schema with the new value', function () {
+      var TestComponent = registerComponent('dummy', {
+        schema: { color: { default: 'red' } }
```
Comment (ngokevin) [inline tests/core/component.test.js]: shouldn't we test with a component that defines `updateSchema`?

File: tests/core/component.test.js [hunk 2]
```diff
@@ -225,4 +225,30 @@ suite('Component', function () {
...
+      });
+      var el = document.createElement('a-entity');
+      var component = new TestComponent(el);
+      component.updateProperties(null);
+      assert.equal(component.data.color, 'red');
+    });
+
+    test('updates the schema with a null value', function () {
+      var TestComponent = registerComponent('dummy', {
+        schema: { color: { default: 'red' } }
```
Comment (ngokevin) [inline tests/core/component.test.js]: going forward, should we stick with spaces-in-objects style? can visually confuse it with block braces, not consistent with arrays, less compact

**Comment by ngokevin** [comment]
I am getting an error with https://github.com/aframevr/aframe/pull/1145 and this PR doesn't fix it.

Mixins + Material = shader not defined: http://codepen.io/team/mozvr/pen/jqMGEq?editors=100

**Comment by ngokevin** [comment]
screen shot 2016-03-13 at 11 40 56 am

**Comment by ngokevin** [comment]
I'll work off of this

**Pull request merged**